### PR TITLE
Example server use cases and other fixes

### PR DESCRIPTION
**Pull request by vicentepinto98**

### CODE REVIEW

File: stackit/internal/services/iaas/networkinterface/resource.go [hunk 1]
```diff
@@ -200,6 +200,7 @@ func (r *networkInterfaceResource) Schema(_ context.Context, _ resource.SchemaRe
 				},
 				PlanModifiers: []planmodifier.String{
 					stringplanmodifier.RequiresReplace(),
+					stringplanmodifier.UseStateForUnknown(),
```
Comment (GokceGK) [inline stackit/internal/services/iaas/networkinterface/resource.go:203]: :+1:

File: docs/resources/server.md [hunk 2]
```diff
@@ -5,38 +5,349 @@ subcategory: ""
...
+    }
+    name              = "example-volume"
+    availability_zone = "eu01-1"
+  }
+  
+  resource "stackit_server" "boot-from-volume" {
+    project_id = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+    name       = "example-server"
+    boot_volume = {
+      size        = 64
```
Comment (gottliebjo) [inline docs/resources/server.md:44]: size is not needed here, as it is stated in the volume

File: examples/resources/stackit_network_area_route/resource.tf [hunk 3]
```diff
@@ -1,4 +1,4 @@
-resource "stackit_network_area" "example" {
+resource "stackit_network_area_route" "example" {
```
Comment (GokceGK) [inline examples/resources/stackit_network_area_route/resource.tf:1]: :+1:

File: stackit/internal/services/iaas/server/const.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
+package server
+
+const markdownDescription = `
+Server resource schema. Must have a region specified in the provider configuration.` + "\n" + `
+~> This resource is in beta and may be subject to breaking changes in the future. Use with caution. See our [guide](https://registry.terraform.io/providers/stackitcloud/stackit/latest/docs/guides/opting_into_beta_resources) for how to opt-in to use beta resources.
+## Example Usage` + "\n" + `
+
+` + "\n```" + `
+
+### Boot from volume` + "\n" +
```
[the remark this answers is not in the document]
Comment (vicentepinto98) [reply stackit/internal/services/iaas/server/const.go:10]: If you render the markdown file, it is actually shown as a header. You can check it by viewing the file here in GitHub, so I think it's good like this

[the remark this answers is not in the document]
Comment (vicentepinto98) [reply stackit/internal/services/iaas/server/const.go:8]: fixed it

**Review by GokceGK — DISMISSED** [review]

File: stackit/internal/services/iaas/networkinterface/resource.go [hunk 5]
```diff
@@ -48,7 +48,6 @@ type Model struct {
 	Name               types.String `tfsdk:"name"`
 	AllowedAddresses   types.List   `tfsdk:"allowed_addresses"`
 	IPv4               types.String `tfsdk:"ipv4"`
-	IPv6               types.String `tfsdk:"ipv6"`
```
Comment (GokceGK) [inline stackit/internal/services/iaas/networkinterface/resource.go:51]: should we remove ipv6 also from datasource

Comment (vicentepinto98) [reply stackit/internal/services/iaas/networkinterface/resource.go:51]: Yes, good catch

**Review by GokceGK — APPROVED** [review]

**Pull request merged**